docs/backend-system/building-backends/08-migrating.md

```diff
@@ -7,3 +7,238 @@ description: How to migrate existing backends to the new backend system
 ---
 
 # Overview
+
+This section describes how to migrate an existing Backstage backend service
+package (typically in `packages/backend`) to use the new backend system.
+
+One of the main benefits of the new backend system is that it abstracts away the
+way that plugins and their dependencies are wired up, leading to a significantly
+simplified backend package that rarely if ever needs to change when plugins or
+their dependencies evolve. You generally don't have to convert all of your
+internal plugins and support classes themselves to the backend system first -
+the migration here will mostly deal with wiring and using compatibility wrappers
+where possible in the backend package itself. We hope that you will find that
+you end up with a much smaller, easier to understand, and easier to maintain
+package as a result of these steps, and then being able to [migrate
+plugins](../building-plugins-and-modules/08-migrating.md) as a separate
+endeavour later.
+
+# Overall Structure
+
+Your typical backend package has a few overall component parts:
+
+- An `index.ts` file that houses all of the creation and wiring together of all
+  of the plugins and their dependencies
+- A `types.ts` file that defines the "environment", i.e. the various
+  dependencies that get created by the backend and passed down into each plugin
+- A `plugins` folder which has one file for each plugin, e.g.
+  `plugins/catalog.ts`
+
+The index file has this overall shape:
+
+```ts
+import todo from './plugins/todo'; // repeated for N plugins
+
+function makeCreateEnv(config: Config) {
+  return (plugin: string): PluginEnvironment => {
+    // ... build per-plugin environment
+  };
+}
+
+async function main() {
+  // ... early init
+  const createEnv = makeCreateEnv(config);
+  const todoEnv = useHotMemoize(module, () => createEnv('todo')); // repeated for N plugins
+  const apiRouter = Router();
+  apiRouter.use('/todo', await todo(todoEnv)); // repeated for N plugins
+  // ... return composite router
```

Comment:
This would be something like "write up and start http server" right?

Reply:
sure, rewritten